libfio.c

```diff
@@ -353,14 +353,17 @@ int initialize_fio(char *envp[])
 	 * can run into problems on archs that fault on unaligned fp
 	 * access (ARM).
 	 */
+	compiletime_assert((offsetof(struct thread_data, ts) % 8) == 0, "ts");
```

Comment:
Because we told the compiler "the ts structure is packed" which means even though ts in within the thread_data struct the compiler doesn't have to care about aligning it. See https://github.com/axboe/fio/issues/356#issuecomment-302919781 for a worked example (alternatively keep the asserts but don't add `__attribute__ ((aligned))` and see if you trigger something.